### PR TITLE
Translate in-band captions to webvtt cues

### PR DESCRIPTION
**Pull request by dmlap**
Pick up caption arrays on transmuxed segments and convert them into WebVTT cues on a programmatically generated captions track.

### CODE REVIEW

File: src/videojs-media-sources.js [hunk 1]
```diff
@@ -247,6 +280,7 @@
       this.pendingBuffers_.length = 0;
 
       // We are no longer in the internal "updating" state
+      this.pendingBuffers_ = [];
```
Comment (imbcmdth) [inline src/videojs-media-sources.js]: No longer needed because I set pendingBuffers_ length to zero above! :)

File: src/videojs-media-sources.js [hunk 2]
```diff
@@ -214,10 +235,12 @@
       var sortedSegments = {
           video: {
             segments: [],
+            captions: [],
             bytes: 0
           },
           audio: {
             segments: [],
+            captions: [],
```
Comment (imbcmdth) [inline src/videojs-media-sources.js]: I though only video can have captions?

File: src/videojs-media-sources.js [hunk 3]
```diff
@@ -37,6 +37,18 @@
           this.settings_.mode === 'html5') {
         self = new window.MediaSource();
         interceptBufferCreation(self);
+
+        // capture the associated player when the MediaSource is
+        // successfully attached
+        self.addEventListener('sourceopen', function() {
+          var video = document.querySelector('[src="' + self.url_ + '"]');
```
Comment (imbcmdth) [inline src/videojs-media-sources.js]: This was a good call by @gkatsev!

**Comment by imbcmdth** [comment]
Minor nits. Otherwise, LGTM

File: src/videojs-media-sources.js (hunk 3 above)
Comment (dmlap) [inline src/videojs-media-sources.js]: Incredibly good call.

Comment (gkatsev) [inline src/videojs-media-sources.js]: lol, so bad.

File: src/videojs-media-sources.js (hunk 2 above)
Comment (dmlap) [inline src/videojs-media-sources.js]: That's true in practice but I thought this simplified the cases downstream.

**Pull request merged**